builtInFunctions/esdtNFTCreate.go

```diff
@@ -351,6 +367,16 @@ func (e *esdtNFTCreate) getESDTNFTCreateInput(
 			return nil, err
 		}
 
+		err = e.validateEsdtType(esdtData.esdtType)
+		if err != nil {
+			return nil, err
+		}
+
+		err = e.validateQuantity(big.NewInt(0).SetBytes(vmInput.Arguments[1]), esdtData.esdtType)
```

Comment:
return quantity on the esdtNFTCreateInput structure. 

And use the information directly from the structure down the line of processing.